menu_gen.rb

```diff
@@ -0,0 +1,59 @@
+# Random Menu Generator - Assignment 2
+# Silver: Use any number of menu items. So if you expand your arrays to
+#   contain 15 or 20 items, your program will show a list of 15 or 20 items.
+# Gold: Expand your solution to ensure that no descriptive term in a menu item is ever repeated.
+#   Expand your solution to allow the user to determine how many items they'd like to see via user input.
+
+# What are you eating?
+type = ["sandwich", "pasta", "panini", "salad", "appetizer",
+  "buffet", "dinner", "sandwich", "popcorn", "steak",
+  "bacon", "watermelon", "peppers", "enchilladas", "fries",
+  "waffles", "pancakes", "pizza", "macaroni and cheese", "smoothy"]
+
+# where is the recipe from/style?
+style = ["cajun", "italian", "mexican", "french", "southern",
+  "chinese", "japanese", "chinese", "indian", "moroccan",
+  "sushi", "finger food", "snack", "breakfast", "elevenses",
+  "fusion", "vegan", "gluten-free", "persian", "local"]
+
+# self-explanatory
+descriptor = ["fresh", "tasty", "crusty", "brittle", "butter",
+  "savory", "sweet", "slimy", "gooey", "salty",
+  "monstrous", "thick", "icey", "steamy", "chocolatey",
+  "appetizing", "braised", "broiled", "caramelized", "soggy"]
+
+# shuffle the arrays up!
+type.shuffle!
+style.shuffle!
+descriptor.shuffle!
+
+# get the user's input on how many menu items they want to see
+puts "Please select between 0 - 20 how many food menu items you would like to see: "
+menu_length = gets.chomp
+
+# this loop will check to make sure user input is a number
+while (menu_length.to_i.to_s != menu_length) && (menu_length.to_f.to_s != menu_length)
```

Comment:
Next time it might be good to add a comment about why you used the `to_i.to_s` code. Someone else reading your code might not know about the conversion of letters to zero